gslib/wildcard_iterator.py

```diff
@@ -583,16 +583,23 @@ class FileWildcardIterator(WildcardIterator):
   files in any subdirectory named 'abc').
   """
 
-  def __init__(self, wildcard_url, ignore_symlinks=False, logger=None):
+  def __init__(self,
+               wildcard_url,
+               exclude_pattern=None,
+               ignore_symlinks=False,
+               logger=None):
     """Instantiates an iterator over BucketListingRefs matching wildcard URL.
 
     Args:
       wildcard_url: FileUrl that contains the wildcard to iterate.
+      exclude_pattern: A regex of paths to ignore during iteration.
```

Comment:
Might we worth mentioning here that it is ignored for Cloud URLs

Reply:
This description is specifically for `FileWildcardIterator` which doesn't work with Cloud URLs.